### PR TITLE
fix(mlx-lm): sorted probs in top_p implementation.

### PR DESCRIPTION
**Pull request by mzbac**
Just realized that the sorted probs indexing introduced a new dimension, so we have to squeeze it out. Also, update the test to cover the shape changes.

### CODE REVIEW

File: llms/mlx_lm/sample_utils.py [hunk 1]
```diff
@@ -22,7 +22,8 @@ def top_p_sampling(logits: mx.array, top_p: float, temperature: float) -> mx.arr
 
     # sort probs in ascending order
     sorted_indices = mx.argsort(probs, axis=-1)
-    sorted_probs = probs[..., sorted_indices]
+    # probs shape is [B,V] and after indexing it will be [1, B, V], so we squeeze it back to [B, V]
+    sorted_probs = probs[..., sorted_indices].squeeze(0)
```
Comment (awni) [inline llms/mlx_lm/sample_utils.py:26]: That doesn't make sense to me, where does it get the `1` from?

```python
>>> import mlx.core as mx
>>> a = mx.zeros((4, 8))
>>> a[..., mx.array([2,3])].shape
(4, 2)
```

Comment (awni) [reply llms/mlx_lm/sample_utils.py:26]: I think I understand now.. it's probably clearer if you do 
```python
sorted_indices = mx.argsort(probs, axis=-1)
probs[..., sorted_indices.squeeze(0)]
```

File: llms/tests/test_sample_utils.py [hunk 2]
```diff
@@ -18,6 +18,7 @@ def test_top_p_sampling(self, mock_categorical):
         expected_top_probs = mx.array([[0.0, 0.0, 0.0, 0.643914]])
         self.assertTrue(mx.allclose(token, expected_token))
         args, _ = mock_categorical.call_args
+        self.assertTrue(args[0].shape == expected_top_probs.shape)
```
Comment (awni) [inline llms/tests/test_sample_utils.py:21]: 👍

File: llms/mlx_lm/sample_utils.py (hunk 1 above)
Comment (mzbac) [reply llms/mlx_lm/sample_utils.py:26]: The shape (1, B, V) is a bit surprising to me. I didn't really understand it, but it seems to be the same behavior in numpy.

**Review by awni — APPROVED** [review]
Thanks!!

**Pull request merged**